CMakeModules/FindOpenCL.cmake

```diff
@@ -1,94 +1,150 @@
-                         "$ENV{ATISTREAMSDKROOT}/include")
-
-      FIND_PATH(OPENCL_INCLUDE_DIRS CL/cl.h
-                PATHS ${OPENCL_LIB_DIR}
-                ENV OpenCL_INCPATH)
-      FIND_PATH(_OPENCL_CPP_INCLUDE_DIRS CL/cl.hpp
-                PATHS ${OPENCL_LIB_DIR}
-                ENV OpenCL_INCPATH)
-
-      # The AMD SDK currently installs both x86 and x86_64 libraries
-      # This is only a hack to find out architecture
-      IF(CMAKE_SIZEOF_VOID_P EQUAL 8)
-          SET(OPENCL_LIB_DIR "$ENV{AMDAPPSDKROOT}/lib/x86_64"
-                             "$ENV{INTELOCLSDKROOT}/lib/x64"
-                             "$ENV{NVSDKCOMPUTE_ROOT}/OpenCL/common/lib/x64"
-                             "$ENV{ATISTREAMSDKROOT}/lib/x86_64")
-      ELSE () # if not 64 bits we assume 32
-          SET(OPENCL_LIB_DIR "$ENV{AMDAPPSDKROOT}/lib/x86"
-                             "$ENV{INTELOCLSDKROOT}/lib/x86"
-                             "$ENV{NVSDKCOMPUTE_ROOT}/OpenCL/common/lib/Win32"
-                             "$ENV{ATISTREAMSDKROOT}/lib/x86")
-      ENDIF()
-
-      FIND_LIBRARY(OPENCL_LIBRARIES OpenCL.lib PATHS ${OPENCL_LIB_DIR} ENV OpenCL_LIBPATH)
-
-  ELSE ()
-
-      # Unix style platforms
-      FIND_LIBRARY(OPENCL_LIBRARIES OpenCL PATHS ENV LD_LIBRARY_PATH ENV OpenCL_LIBPATH)
-
-      GET_FILENAME_COMPONENT(OPENCL_LIB_DIR ${OPENCL_LIBRARIES} PATH)
-      GET_FILENAME_COMPONENT(_OPENCL_INC_CAND ${OPENCL_LIB_DIR}/../../include ABSOLUTE)
-
-      # The AMD SDK currently does not place its headers
-      # in /usr/include, therefore also search relative
-      # to the library
-      FIND_PATH(OPENCL_INCLUDE_DIRS CL/cl.h PATHS ${_OPENCL_INC_CAND} "/usr/local/cuda/include" "/opt/AMDAPP/include" ENV OpenCL_INCPATH)
-      FIND_PATH(_OPENCL_CPP_INCLUDE_DIRS CL/cl.hpp PATHS ${_OPENCL_INC_CAND} "/usr/local/cuda/include" "/opt/AMDAPP/include" ENV OpenCL_INCPATH)
-
-  ENDIF ()
-
-ENDIF ()
-
-FIND_PACKAGE_HANDLE_STANDARD_ARGS(OpenCL DEFAULT_MSG OPENCL_LIBRARIES OPENCL_INCLUDE_DIRS)
-
-IF(_OPENCL_CPP_INCLUDE_DIRS)
-        SET( OPENCL_HAS_CPP_BINDINGS TRUE )
-        LIST( APPEND OPENCL_INCLUDE_DIRS ${_OPENCL_CPP_INCLUDE_DIRS} )
-        # This is often the same, so clean up
-        LIST( REMOVE_DUPLICATES OPENCL_INCLUDE_DIRS )
-ENDIF(_OPENCL_CPP_INCLUDE_DIRS)
-
-MARK_AS_ADVANCED(
-  OPENCL_INCLUDE_DIRS
-  OPENCL_LIBRARIES
-)
 
+#=============================================================================
+# Copyright 2014 Matthaeus G. Chajdas
+#
+# Distributed under the OSI-approved BSD License (the "License");
+# see accompanying file Copyright.txt for details.
+#
+# This software is distributed WITHOUT ANY WARRANTY; without even the
+# implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the License for more information.
+#=============================================================================
+# (To distribute this file outside of CMake, substitute the full
+#  License text for the above reference.)
```

Comment:
From where you found FindOpenCL.cmake.  I guess from the text that it is bundled with cmake past a certain version, so the copyright file it refers to is probably in the cmake sources.